builtin/providers/triton/resource_machine.go

```diff
@@ -0,0 +1,436 @@
+	machineStateChangeCheckInterval = 10 * time.Second
+
+	resourceMachineMetadataKeys = map[string]string{
+		// semantics: "schema_name": "metadata_name"
+		"root_authorized_keys": "root_authorized_keys",
+		"user_script":          "user-script",
+		"user_data":            "user-data",
+		"administrator_pw":     "administrator-pw",
+	}
+)
+
+func resourceMachine() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMachineCreate,
+		Exists: resourceMachineExists,
+		Read:   resourceMachineRead,
+		Update: resourceMachineUpdate,
+		Delete: resourceMachineDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description:  "friendly name",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: resourceMachineValidateName,
+			},
+			"type": {
+				Description: "machine type (smartmachine or virtualmachine)",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"state": {
+				Description: "current state of the machine",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"dataset": {
+				Description: "dataset URN the machine was provisioned with",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"memory": {
+				Description: "amount of memory the machine has (in Mb)",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"disk": {
+				Description: "amount of disk the machine has (in Gb)",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"ips": {
+				Description: "IP addresses the machine has",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"tags": {
+				Description: "machine tags",
+				Type:        schema.TypeMap,
+				Optional:    true,
+			},
+			"created": {
+				Description: "when the machine was created",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"updated": {
+				Description: "when the machine was update",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"package": {
+				Description: "name of the package to use on provisioning",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"image": {
+				Description: "image UUID",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				// TODO: validate that the UUID is valid
+			},
+			"primaryip": {
+				Description: "the primary (public) IP address for the machine",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"networks": {
+				Description: "desired network IDs",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				// TODO: this really should ForceNew but the Network IDs don't seem to
+				// be returned by the API, meaning if we track them here TF will replace
+				// the resource on every run.
+				// ForceNew:    true,
```

Comment:
This comment doesn't sound like it's describing `ForceNew` but rather `Computed`?